In [1]:
try:
    from transformers import pipeline
except:
    !pip install transformers datasets

我們剛剛介紹的情感分析可以被認為是文本分類的一種特殊情況，其中類別（或類別）只有正面、負面或中性。文本分類更通用，因為它可以將傳入文本（例如句子、段落或文檔）分類（或分類）到預定義的類別中。讓我們看看這在實踐中意味著什麼。

In [2]:
# Specify model
model = 'facebook/bart-large-mnli'

# Specify Task
task = 'zero-shot-classification'

# Specify input text
input_text = 'This is a tutorial about using pre-trained models through HuggingFace'

# Identify the classes/categories/labels
labels = ['business', 'sports', 'education', 'politics', 'music']

# Instantiate pipeline
classifier = pipeline(task, model = model)

# Store the output of the analysis
output = classifier(input_text, candidate_labels = labels)

# Return output
output

C:\Users\cti110016\AppData\Local\anaconda3\Lib\site-packages\huggingface_hub\file_download.py:133: UserWarning: `huggingface_hub` cache-system uses symlinks by default to efficiently store duplicated files but your machine does not support them in C:\Users\cti110016\.cache\huggingface\hub. Caching files will still work but in a degraded version that might require more space on your disk. This warning can be disabled by setting the `HF_HUB_DISABLE_SYMLINKS_WARNING` environment variable. For more details, see https://huggingface.co/docs/huggingface_hub/how-to-cache#limitations.
To support symlinks on Windows, you either need to activate Developer Mode or to run Python as an administrator. In order to see activate developer mode, see this article: https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development
  warnings.warn(message)


{'sequence': 'This is a tutorial about using pre-trained models through HuggingFace',
 'labels': ['education', 'business', 'music', 'sports', 'politics'],
 'scores': [0.40113669633865356,
  0.2170695662498474,
  0.1454731822013855,
  0.1450750082731247,
  0.09124555438756943]}

結果非常有趣！分數對應於每個標籤，從最大到最小排序以便於閱讀。例如，結果表明我們的句子被標記為“教育”，得分約為 40%，其次是“商業”，得分約為 22%，而“音樂”、“體育”和“政治”的標籤得分非常低分數，這對我來說總體來說是有意義的。